src/src/pages/capabilities/KafkaCluster/Topic.js

```diff
@@ -172,9 +190,20 @@ export default function Topic({ topic, isSelected, onHeaderClicked }) {
     });
   };
 
+  const fetchAndSortMessageContractsForTopic = async (topic) => {
+    const result = await selfServiceApiClient.getMessageContracts(topic);
+    result.sort((a, b) => a.messageType.localeCompare(b.messageType));
+    return result;
+  };
+
   const handleAddMessageContract = async (formValues) => {
     await addMessageContractToTopic(topic.kafkaClusterId, topic.id, formValues);
     setShowMessageContractDialog(false);
+    setIsLoadingContracts(true);
+    await sleep(1000);
```

Comment:
Using sleep is usually not a good sign.
What are we trying to achieve here?

My guess is we want to make sure the frontend is updated with the latest message topics, but the server may not be updated.

I would suggest some version of continuous polling as we do elsewhere (i.e. Keep asking every X seconds, until the response contains what I expect)
OR
A pretend-solution (i.e. The server might not be ready, but I have all the data, so I just add it to the list in the frontend)